core/src/main/scala/org/apache/spark/rdd/AsyncRDDActions.scala

```diff
@@ -84,18 +87,18 @@ class AsyncRDDActions[T: ClassTag](self: RDD[T]) extends Serializable with Loggi
       } else {
         // The number of partitions to try in this iteration. It is ok for this number to be
         // greater than totalParts because we actually cap it at totalParts in runJob.
-        var numPartsToTry = 1L
+        var numPartsToTry = Math.max(self.conf.get(RDD_LIMIT_INITIAL_NUM_PARTITIONS), 1)
```

Comment:
Enforce it in Config itself and always use `self.conf.get(RDD_LIMIT_INITIAL_NUM_PARTITIONS)` ?

For RDD_LIMIT_INITIAL_NUM_PARTITIONS:
```
...
.intConf
.checkValue(_ > 0, "")
...
```

Reply:
nice idea, modified accordingly